core/execution/common/market_activity_tracker.go

```diff
@@ -593,7 +596,9 @@ func (mat *MarketActivityTracker) isEligibleForReward(asset, party string, marke
 		}
 	}
 	if !notionalTimeWeightedAveragePositionRequired.IsZero() {
-		if mat.getTWNotionalPosition(asset, party, markets).LT(notionalTimeWeightedAveragePositionRequired) {
+		notional := mat.getTWNotionalPosition(asset, party, markets)
```

Comment:
ok, there's a few problem here (partly my fault):
1. so the isEligibleForReward will be called possibly a lot more times than you need an event for so maybe emitting the event as it is just now is not a good idea (that's on me)
2. note that the function is called with a set of markets - so what it means is that this is the tw notional for the party for the asset *in this set of markets* - now in the event you're ignoring this context so it gets lost and you will emit the same value regardless of the set of markets which I think is wrong - I think if you were including the markets in the 
3. I think we might want to emit an event with 0 for a party with 0 as its still informative.

maybe instead of or in addition to party and asset, just add the game_id (as hash(dispatch strategy) to the event - this would fix problem 2 and possible 1 as well.

Reply:
Added the game id as an additional grouping for the time weighted notional position as suggested and updated the API accordingly.